src/HMRServer.js

```diff
@@ -6,15 +6,63 @@ class HMRServer {
       this.wss = new WebSocket.Server({port: 0}, resolve);
     });
 
+    this.wss.on('connection', (ws) => {
+      if (this.unresolvedError) {
+        ws.send(JSON.stringify(this.unresolvedError))
+      }
+    });
+
     return this.wss._server.address().port;
   }
 
   stop() {
     this.wss.close();
   }
 
+  emitError(err) {
+    let message = typeof err === 'string' ? err : err.message;
+    if (!message) {
+      message = 'unknown error'
+    }
+
+    if (err.fileName) {
+      let fileName = err.fileName;
+      if (err.loc) {
+        fileName += `:${err.loc.line}:${err.loc.column}`;
+      }
+
+      message = `${fileName}: ${message}`;
+    }
+
+    let stack
+    if (err.codeFrame) {
+      stack = err.codeFrame;
+    } else if (err.stack) {
+      stack = err.stack.slice(err.stack.indexOf('\n') + 1);
+    }
+    
+    // store the most recent error so we can notify new connections
+    // and so we can broadcast when the error is resolved
+    this.unresolvedError = {
+      type: 'error',
+      error: {
+        message,
+        stack
+      }
+    }
+
+    this.broadcast(this.unresolvedError)
+  }
+
   emitUpdate(assets) {
-    let msg = JSON.stringify({
+    if (this.unresolvedError) {
+      this.unresolvedError = null
+      this.broadcast({
```

Comment:
Initially tracked this in the `hmr-runtime`, but when I realized it needed to be tracked here too I thought it might make more sense to keep the event handling in `hmr-runtime` mostly stateless.